diff --git a/imagecrop.module b/imagecrop.module
index 8287653..858c33a 100755
--- a/imagecrop.module
+++ b/imagecrop.module
@@ -470,15 +470,10 @@ function get_imagecrop_styles($enabled_styles = array()) {
       if ($effect['name'] == 'imagecrop_javascript') {
 
         // Skip if current style is not enabled.
-        if ($apply_filter && !in_array($machine_name, $enabled_styles)) {
-
-          // Make sure old versions still work.
-          if (!isset($style['isid']) || !in_array($style['isid'], $enabled_styles)) {
-            continue;
-          }
-
+        if ($apply_filter && !in_array($style['isid'], $enabled_styles)) {
+          continue;
         }
-        $styles[$machine_name] = $style['name'];
+        $styles[$style['isid']] = $style['name'];
 
       }
     }
