--- imagecrop.module	2011-04-26 11:24:28.263820639 +0300
+++ imagecrop.module.new	2011-04-26 11:28:02.399820072 +0300
@@ -63,6 +63,9 @@ function imagecrop_theme() {
       'variables' => array('data' => NULL),
       'file' => 'includes/imagecrop.theme.inc',
     ),
+    'imagecrop_reuse' => array(
+      'variables' => array('element' => NULL),
+    )
   );
 
 }
@@ -122,6 +125,14 @@ function imagecrop_image_effect_info() {
     'form callback' => 'imagecrop_effect_form',
     'summary theme' => 'imagecrop_effect_summary',
   );
+  
+  $effects['imagecrop_reuse'] = array(
+      'label' => t('Javascript crop reuse selection'),
+      'help' => 'Reuse crop selection from another preset.',
+      'effect callback' => 'imagecrop_reuse_image',
+      'form callback' => 'imagecrop_reuse_form',
+      'summary theme' => 'theme_imagecrop_reuse',
+     );
 
   return $effects;
 
@@ -559,4 +570,41 @@ function imagecrop_colorbox($url, $width
 
   return '[<a class="colorbox-load" href="'. $url .'?iframe=true&height='. $height .'&width='. $width .'">'. t('Crop this image') .'</a>]';
 
+}
+
+function imagecrop_reuse_form($data) {
+  $presets = get_imagecrop_styles();//imagecrop_presets_list();
+  if (count($presets) > 0) {
+    $form['imagecrop_presetname'] = array(
+      '#title' => t('Enabled imagecrop presets'),
+      '#type' => 'select',
+      '#options' => $presets,
+      '#default_value' => (array) $data['imagecrop_presetname'],
+    );
+  }
+  else {
+    $form['imagecrop_warning'] = array(
+      '#value' => t('No preset is found with the javascript_crop action so far. If you want to take advantage of this action, you will need to create at least one preset with javascript_crop action.'),
+    );
+  } 
+  return $form;
+}
+
+function theme_imagecrop_reuse($element) {
+  $data = $element['#value'];
+  $description = $data['imagecrop_presetname'];
+  return $description;
+}
+
+function imagecrop_reuse_image(&$image, $data) {
+  if ($preset_name = $data['imagecrop_presetname']) {
+    $preset = imagecache_preset_by_name($preset_name);
+    foreach ($preset['actions'] as $action) if($action['action'] == 'imagecrop_javascript') {
+      $GLOBALS['imagecrop_presetid'] = $preset['presetid'];
+      _imagecache_apply_action($action, $image);
+      return TRUE;
+    } 
+    return FALSE;
+  }
+  return FALSE;
 }
\ No newline at end of file
