diff --git a/includes/imagecrop.admin.inc b/includes/imagecrop.admin.inc
index ae3d663..020efcf 100755
--- a/includes/imagecrop.admin.inc
+++ b/includes/imagecrop.admin.inc
@@ -201,8 +201,8 @@ function imagecrop_cropping_page($fid, $style_name, $entity_type = NULL, $bundle
       throw new Exception("You don't have permissions to crop this image.");
     }
 
-    $imagecrop->loadCropSettings();
     $imagecrop->writeCropreadyImage();
+    $imagecrop->loadCropSettings();
     $imagecrop->addImagecropUi(TRUE);
 
     return theme(($imagecrop->extraControls ? 'imagecrop_crop_ui_advanced' : 'imagecrop_crop_ui'), array(
diff --git a/includes/imagecrop.class.inc b/includes/imagecrop.class.inc
index 0af9a60..aabb933 100755
--- a/includes/imagecrop.class.inc
+++ b/includes/imagecrop.class.inc
@@ -327,9 +327,11 @@ class ImageCrop {
    */
   public function loadCropSettings() {
 
-    $size = getimagesize(drupal_realpath($this->file->uri));
-    $this->imageWidth = $this->originalImageWidth = $size[0];
-    $this->imageHeight = $this->originalImageHeight = $size[1];
+    if (!$this->inCroppingMode) {
+      $size = getimagesize($this->file->uri);
+      $this->imageWidth = $this->originalImageWidth = $size[0];
+      $this->imageHeight = $this->originalImageHeight = $size[1];
+    }
 
     $settings = db_select('image_crop_settings')
       ->fields('image_crop_settings')
@@ -477,10 +479,15 @@ class ImageCrop {
     $style = $this->imageStyle;
 
     foreach ($this->imageStyle['effects'] as $key => $effect) {
+
       if ($effect['name'] == 'imagecrop_javascript') {
         $unset = TRUE;
       }
-      unset($style['effects'][$key]);
+
+      if ($unset) {
+        unset($style['effects'][$key]);
+      }
+
     }
 
     if ($this->scale !== 'original') {
@@ -494,6 +501,11 @@ class ImageCrop {
 
     image_style_create_derivative($style, $this->file->uri, $this->cropDestination);
 
+    // Set imageWidth and height after effects are done.
+    $size = getimagesize($this->cropDestination);
+    $this->imageWidth = $this->originalImageWidth = $size[0];
+    $this->imageHeight = $this->originalImageHeight = $size[1];
+
   }
 
   /**
diff --git a/js/imagecrop.ui.crop.js b/js/imagecrop.ui.crop.js
index 39c6662..eeb9cf0 100755
--- a/js/imagecrop.ui.crop.js
+++ b/js/imagecrop.ui.crop.js
@@ -36,11 +36,12 @@ Drupal.Imagecrop.cropUi.initControls = function() {
   
   Drupal.Imagecrop.resizeMe = $('#resizeMe');
   Drupal.Imagecrop.cropUi.cropContainer = $('#image-crop-container');
+  Drupal.Imagecrop.cropUi.cropWrapper = $('#imagecrop-crop-wrapper');
   
   if (Drupal.Imagecrop.resizeMe.resizable) { 
 
     Drupal.Imagecrop.resizeMe.resizable({
-      containment: Drupal.Imagecrop.cropUi.cropContainer,
+      containment: Drupal.Imagecrop.cropUi.cropWrapper,
       aspectRatio: Drupal.settings.imagecrop.resizeAspectRatio,
       autohide: true,
       handles: 'n, e, s, w, ne, se, sw, nw',
@@ -51,7 +52,7 @@ Drupal.Imagecrop.cropUi.initControls = function() {
 
   Drupal.Imagecrop.resizeMe.draggable({
     cursor: 'move',
-    containment: Drupal.Imagecrop.cropUi.cropContainer,
+    containment: Drupal.Imagecrop.cropUi.cropWrapper,
     drag: Drupal.Imagecrop.cropUi.dragListener
   });
   
diff --git a/reuse_effect-1244996-8.patch b/reuse_effect-1244996-8.patch
deleted file mode 100644
index b63c53c..0000000
--- a/reuse_effect-1244996-8.patch
+++ /dev/null
@@ -1,33 +0,0 @@
-diff --git a/includes/imagecrop.effects.inc b/includes/imagecrop.effects.inc
-index c32bd42..3a78ded 100644
---- a/includes/imagecrop.effects.inc
-+++ b/includes/imagecrop.effects.inc
-@@ -201,11 +201,11 @@ function imagecrop_reuse_form($data) {
-   }
- 
-   if (count($presets) > 0) {
--    $form['imagecrop_sid'] = array(
-+    $form['imagecrop_style'] = array(
-       '#title' => t('Use the crop settings from'),
-       '#type' => 'select',
-       '#options' => $presets,
--      '#default_value' => isset($data['imagecrop_sid']) ? $data['imagecrop_sid'] : '',
-+      '#default_value' => isset($data['imagecrop_style']) ? $data['imagecrop_style'] : '',
-     );
-   }
-   else {
-@@ -228,12 +228,12 @@ function imagecrop_reuse_form($data) {
-  */
- function imagecrop_reuse_effect(&$image, $data) {
- 
--  if (empty($data['imagecrop_sid'])) {
-+  if (empty($data['imagecrop_style'])) {
-     return FALSE;
-   }
- 
-   // Load selected image style and apply the imagecrop_javascript action.
--  $style = image_style_load(NULL, $data['imagecrop_sid']);
-+  $style = image_style_load($data['imagecrop_style']);
-   foreach ($style['effects'] as $effect) {
-     if ($effect['name'] == 'imagecrop_javascript') {
-       $GLOBALS['imagecrop_style'] = $style['name'];
